docs/rules/no-empty-pattern.md

```diff
@@ -0,0 +1,42 @@
+# No empty pattern (no-empty-pattern)
+
```

Comment:
We need more explanation here about why empty patterns are not a good idea. Also we need explanation as to what this rule is trying to achieve.